apps/condo/domains/miniapp/components/ConnectedAppsWithIconsProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import { SortB2BAppContextsBy } from '@app/condo/schema'
+import get from 'lodash/get'
+import React, { createContext, useContext, useState } from 'react'
+
+import { useDeepCompareEffect } from '@open-condo/codegen/utils/useDeepCompareEffect'
+import { useOrganization } from '@open-condo/next/organization'
+
+import { ALL_MENU_CATEGORIES, DEFAULT_MENU_CATEGORY } from '@condo/domains/common/constants/menuCategories'
+import { CONTEXT_FINISHED_STATUS } from '@condo/domains/miniapp/constants'
+import { B2BAppContext } from '@condo/domains/miniapp/utils/clientSchema'
+
+import type { B2BAppContext as ContextType } from '@app/condo/schema'
+
+type ContextsByCategories = Record<string, Array<ContextType>>
+
+type IConnectedAppsWithIconsContext = {
+    contextsByCategories: ContextsByCategories
+    connectedAppsIds: Array<string>
+    refetch: () => void
+}
+
+export const ConnectedWithIconsContext = createContext<IConnectedAppsWithIconsContext>({
+    contextsByCategories: {},
+    connectedAppsIds: [],
+    refetch: () => ({}),
+})
+
+export const ConnectedAppsWithIconsContextProvider: React.FC = ({ children }) => {
+    const { organization } = useOrganization()
+    const orgId = get(organization, 'id', null)
+    const [contextsByCategories, setContextsByCategories] = useState<ContextsByCategories>({})
+    const [connectedApps, setConnectedApps] = useState<Array<string>>([])
+
+    const { objs, refetch } = B2BAppContext.useObjects({
+        where: {
+            organization: { id: orgId },
+            app: { icon_not: null },
+            status: CONTEXT_FINISHED_STATUS,
+        },
+        sortBy: [SortB2BAppContextsBy.CreatedAtAsc],
+    })
+
+    useDeepCompareEffect(() => {
+        const connectedAppsIds: Array<string> = []
+        const contextsByCategories: ContextsByCategories = Object.assign({}, ...ALL_MENU_CATEGORIES.map(category => ({ [category]: [] })))
+        for (const ctx of objs) {
+            const appCategory = get(ctx, ['app', 'menuCategory'], DEFAULT_MENU_CATEGORY) || DEFAULT_MENU_CATEGORY
```

Comment:
If you have a server response like `const a = { field: null }` and you call get like  `const b = get(a, 'field', default)` you will receive null. Since fieldPath is exist on `a` object and has a value.